qa-pipelines/qa-pipeline.yml

```diff
@@ -99,6 +99,11 @@ jobs:
         MAGIC_DNS_SERVICE: ((magic-dns-service))
         CAP_INSTALL_VERSION: ((cap-sle-url))
         ENABLE_CF_DEPLOY_PRE_UPGRADE: ((enable-cf-deploy-pre-upgrade))
+        AZ_SP_APPID: ((az-sp-app-id))
```

Comment:
To limit the number of parameters that need to be passed in, see how we use the GKE private key information.

Reply:
I see what you are saying
```
    base64 -d <<< "${GKE_PRIVATE_KEY_BASE64}" > gke-key.json
    export GKE_SERVICE_ACCOUNT_EMAIL=$(jq -r .client_email gke-key.json)
    export GKE_PROJECT_ID=$(jq -r .project_id gke-key.json)
```
The way I see it. its not about how we are doing things in cf-ci repo, it more about how we doing thing in cloudfoundary/secure. All the credential are stored in this way. I guess it because its easier for people to know what we have in there.

Reply:
We control the format of the secrets in the gpg-encrypted secrets file used when deploy the pipeline, putting all the credentials needed for a given platform into one key in that secrets file means that we have fewer parameters we have to remember to include

Reply:
won't be doing this, as per our discussion